src/pages/drivers/[driver].tsx

```diff
@@ -69,10 +69,105 @@ const individualDriverSchema = z.object({
   season: z.string(),
 })
 
-function driverWins() {
-  return
+const resultsSchema = z.object({
+  Constructor: z.object({
+    constructorId: z.string(),
+    name: z.string(),
+    nationality: z.string(),
+  }),
+  Driver: z.object({
+    code: z.string(),
+    dateOfBirth: z.string(),
+    driverId: z.string(),
+    familyName: z.string(),
+    givenName: z.string(),
+    nationality: z.string(),
+    permanentNumber: z.string(),
+  }),
+  FastestLap: z.object({
+    lap: z.string(),
+    rank: z.string(),
+  }),
+  grid: z.string(),
+  laps: z.string(),
+  number: z.string(),
+  points: z.string(),
+  position: z.string(),
+  positionText: z.string(),
+  status: z.string(),
+})
+
+const racesSchema = z.object({
+  Circuit: z.object({
+    Location: z.object({
+      country: z.string(),
+      lat: z.string(),
+      locality: z.string(),
+      long: z.string(),
+    }),
+  }),
+  Results: z.array(resultsSchema),
+  date: z.string(),
+  raceName: z.string(),
+  round: z.string(),
+  season: z.string(),
+})
+
+
+
+
+
+
+
+
+type Races = z.infer<typeof racesSchema>
+type Kind = 'position' | 'grid'
+
+function getDriverStats(
+  results: Races[],
+  key: Kind,
+  value: number | string
+) {
+  return results?.reduce((total, current) => {
+    if (current?.Results[0][key] === value) {
+      total += 1
+    }
+    return total
+  }, 0)
 }
 
+type Races1 = z.infer<typeof racesSchema>
+type Kind1 = 'rank'
+type Kind2 = 'points'
+
+function getDriverStats1(
+  results: Races1[],
+  key: Kind1,
+  value: number | string
+) {
+  return results?.reduce((total, current) => {
+    if (current?.Results[0]['status'] != "Withdrew"){
```

Comment:
Totally up to you, but you could just use dot notation for consistency

```suggestion
    if (current?.Results[0].status != "Withdrew") {
```

Reply:
Not Done